lib/rules/no-unused-state.js

```diff
@@ -237,6 +237,26 @@ module.exports = {
         ) {
           addStateFields(node.value);
         }
+
+        if (
+          !node.static &&
+          node.value &&
+          node.value.type === 'ArrowFunctionExpression'
+        ) {
+          // Create a new set for this.state aliases local to this function.
+          classInfo.aliases = new Set();
```

Comment:
i'm not seeing anything that adds to this set?

If it otherwise crashes because this doesn't exist; then let's initialize it in every case.

Reply:
This list will only be added to if the property name gets assigned to a different variable name.  

For instance,

```js
class App extends React.Component {
        state = { foo: 0 };

        doSomething = () => {
          const { state: { foo: bar }} = this;

          return bar;
        }

        render() {
          return <SomeComponent />;
        }
      }
```

will add `'bar'` to the list. I have added one more test regarding this.

I also refactored how the code deals with `classInfo.aliases` according to your feedback. We still need to clear the set after each method definition though because each alias added is local to the class method. Let me know if this looks better.